### DISCUSSION TITLE
Networkx error when using les_miserables_graph

### DISCUSSION THREAD

**Discussion by fakeNka151203** [networkx/networkx#7723] [Q&A] [answered]
les_miserables_graph reports an error that 

`File "D:\Minicoda\envs\d2l\lib\site-packages\networkx\classes\graph.py", line 1356, in neighbors raise NetworkXError(f"The node {n} is not in the graph .") from err networkx.exception.NetworkXError: The node 1 is not in the graph.`

I am reproducing the code in this link: https://github.com/yaoweihu/Fairness-in-RL?utm_source=catalyzex.com

When using karate_club_graph(), the source code still works fine, however when using les_miserables_graph(), the error is reported as above. I am confused because I do not know if the author of the above article made adjustments just to fit karate_club_graph() or not?

**Comment by dschult** [comment, marked as answer]
The `les_miserables_graph` does not have integers as nodes. So there is no expectation that the number 1 should be a node. If the code you are looking at specifies nodes, you will need to adjust which nodes it refers to when you rewrite the code to use a new graph.

**Reply by rossbar** [reply]
You can see which nodes are available with the `nodes` attribute:

```python
>>> G = nx.les_miserables_graph()
>>> G.nodes
```

If you prefer integer nodes, you might want to check out [`nx.convert_node_labels_to_integers`](https://networkx.org/documentation/stable/reference/generated/networkx.relabel.convert_node_labels_to_integers.html)
